### PR TITLE
Add enable ASM egress gateway commands for az aks mesh

### PR DESCRIPTION
**Pull request by nshankar13**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by nshankar13** [comment]
@andyliuliming @FumingZhang @zhoxing-ms @yanzhudd @yonzhan need some urgent reviews on this as we are trying to meet 9/19 deadline for CLI changes and to get this into the GA API changes in `azure-cli/acs` as well.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -7413,6 +7413,68 @@ def test_aks_azure_service_mesh_with_ingress_gateway(self, resource_group, resou
             self.is_empty(),
         ])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_azure_service_mesh_with_egress_gateway(self, resource_group, resource_group_location):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:7418]: Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=79813556&view=results) to validate the change.

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:7418]: test passed

**Comment by yanzhudd** [comment]
It is suggested to add some details of changes in https://github.com/Azure/azure-cli-extensions/pull/6768#issue-1898959467

**Comment by yanzhudd** [comment]
Please add change logs in HISTORY.rst. If you want to release a new version, please change version in setup.py file.

**Comment by nshankar13** [comment]
@FumingZhang @yanzhudd updated the description.

**Pull request merged**